shoop/xtheme/plugins/widgets.py

```diff
@@ -90,3 +91,38 @@ def decompress(self, value):
         else:
             out_dict[FALLBACK_LANGUAGE_CODE] = value
         return out_dict
+
+
+class XThemeModelChoiceWidget(forms.Select):
+    def render(self, name, value, attrs=None, choices=()):
+        return mark_safe(
+            render_to_string("shoop/xtheme/_model_widget.jinja", {
+                "name": name,
+                "selected_object": [object for pk, object in self.choices if pk == int(value)][0] if value else None,
+                "objects": self.choices,
+            })
+        )
+
+
+class XThemeModelChoiceField(forms.ModelChoiceField):
+    def __init__(self, queryset, empty_label="---------", required=True, label=None, initial=None, *args, **kwargs):
+        super(XThemeModelChoiceField, self).__init__(
+            queryset,
+            empty_label=empty_label,
+            required=required,
+            widget=XThemeModelChoiceWidget,
+            label=label,
+            initial=initial,
+            *args,
+            **kwargs
+        )
+
+    def _get_custom_choices(self):
+        return XThemeModelChoiceIterator(self)
+
+    choices = property(_get_custom_choices, forms.ChoiceField._set_choices)
+
+
+class XThemeModelChoiceIterator(forms.models.ModelChoiceIterator):
```

Comment:
It's not I think that overriding label_from_instance could be better solution for this.